gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/AbstractTaskSpec.groovy

```diff
@@ -0,0 +1,95 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.tasks
+
+import java.nio.file.Files
+import com.palantir.gradle.dist.GradleIntegrationSpec;
+import java.nio.file.StandardCopyOption;
+import nebula.test.dependencies.DependencyGraph;
+import nebula.test.dependencies.GradleDependencyGenerator;
+
+abstract class AbstractTaskSpec extends GradleIntegrationSpec {
```

Comment:
I would really prefer avoiding a test class hierarchy and instead pull things out into shared utilities. Otherwise it can be very confusing to understand the setup of any given test and inadvertently break the tests assumptions

Reply:
replaced with delegate fixture